src/rust/src/builder.rs

```diff
@@ -1,14 +1,33 @@
-use lyon::{geom::euclid::UnknownUnit, math::point, path::traits::PathBuilder};
-use ttf_parser::{colr::Painter, Face};
+use std::collections::HashMap;
+
+use lyon::{
+    geom::euclid::UnknownUnit,
+    math::point,
+    path::{
+        traits::{Build, PathBuilder},
+        Path,
+    },
+};
+use ttf_parser::{
+    colr::{Paint, Painter},
+    Face, NormalizedCoordinate, RgbaColor,
+};
 
 pub struct LyonPathBuilder {
     // It's not very elegant to store the glyph ID (not the `glyphId` ttf-parser
     // uses, but the glyph count) and path ID in attributes, but it seems the
     // attribute is the only thing we can pass to tessellators.
-    pub builder: lyon::path::builder::Transformed<
-        lyon::path::path::BuilderWithAttributes,
-        lyon::math::Transform,
+    pub builders: Vec<
+        lyon::path::builder::Transformed<
+            lyon::path::path::BuilderWithAttributes,
+            lyon::math::Transform,
+        >,
     >,
+    // one layer has only one color
+    pub layer_color: HashMap<usize, RgbaColor>,
```

Comment:
Note to self: since `paint()` is invoked AFTER `outline_glyph()`, it cannot be treated as an attribute of `lyon::path::Path::builder_with_attributes()`. Record the color of the layer here and use it afterwards.